feat: dgfip and inpi bce finance data comparison

rmonnier9 opened the pull request:
- Fonctionnalité
- Détails :
  - Comparaison entre INPI/BCE et DGFiP pour les données financières

#1512 

Sélecteur de cas d'usage sans données open data :
<img width="1278" alt="image" src="https://github.com/user-attachments/assets/76a91f87-84c5-4409-b869-641e343dfbc1" />

Sélecteur de cas d'usage avec données open data :
<img width="1261" alt="image" src="https://github.com/user-attachments/assets/e5e15891-9cd4-42b3-9304-45c00726be26" />

Données différente :
<img width="1332" alt="image" src="https://github.com/user-attachments/assets/ac51cafd-293b-44a1-872e-519a75b55d7a" />

XavierJp commented:
Alors en premier visionnage, purement visuel / UX : 
- que fait-on de 2017 ?
- Incohérence DGFIP est peut etre trop gros. Peut être juste `incohérence` et ensuite on met le détail dans le tooltip ?

XavierJp commented on hooks/fetch/use-API-route-data.ts:
```diff
@@ -58,6 +58,35 @@ export function useAPIRouteData<T extends APIRoutesPaths>(
   return response;
 }
 
+export function useManualAPIRouteData<T extends APIRoutesPaths>(
+  route: T,
+  slug: string,
+  session: ISession | null
+) {
+  const [response, setResponse] = useState<
+    IDataFetchingState | RouteResponse<T>
+  >(IDataFetchingState.LOADING);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const fetchData = useCallback(
+    async (options: Options<T>) => {
+      setIsLoading(true);
+      const result = await fetchAPIRoute<T>(route, slug, session, options);
+      if (result) {
+        setResponse(result);
+      }
+      setIsLoading(false);
```
I dont understand what this is doing

XavierJp reviewed (approved):
I think current implementation is already complex and. we have to add Banque de France. We might need a stronger abstraction on this

XavierJp commented:
There is an issue, Danone for instance has both bilan consolidé (entire Danone groupe) and bilan complet (Danone uniteLegale). And DGFiP CA numbers looks like the bilan complet ones.

I suggest to split the two : 
- a bilan consolidé dedicated section, open data only
- a bilan complet or simplifié one, open data by default and eventually with CA
- we also need a summary in this page. It has too many sections

XavierJp commented:
@rmonnier9 I reviewed and I suggest a full revamp of the données financieres page : 
- add a summary
- differentiate bilans Consolidé from the rest when relevant, in both the bilans PDF section from Inpi and in Indicateurs financiers from ODS
- changed a lot of names bilans -> became indicateursFinanciers, and bilans became bilansBDF depending on context
- I only did the merge for the bilans that are not "Consolidés"
- I also fixed a bug with the `WithIgnoreBot` that would trigger when you wait a bit too long to click on `<UseCaseSelector />`

I see at least two possible improvements with the Chiffre d’Affaires API, that we should share with API Entreprise : 
- we lack the bilan type (complet, simplifié etc.)
- we dont have any CA older than 3 years ago

This revamp include pretty heavy changes to the open data page and I think we can communicate about it @HAEKADI 
![image](https://github.com/user-attachments/assets/b8cb9db5-1af2-4e72-8909-e5f9290b7d91)

rmonnier9 commented:
> @rmonnier9 I reviewed and I suggest a full revamp of the données financieres page :
> 
> * add a summary
> * differentiate bilans Consolidé from the rest when relevant, in both the bilans PDF section from Inpi and in Indicateurs financiers from ODS
> * changed a lot of names bilans -> became indicateursFinanciers, and bilans became bilansBDF depending on context
> * I only did the merge for the bilans that are not "Consolidés"
> * I also fixed a bug with the `WithIgnoreBot` that would trigger when you wait a bit too long to click on `<UseCaseSelector />`
> * I also drew another lineChart for DGFiP CA and added a dedicated line. UX-wise it is simpler than trying to fit it the same cell as the opendata CA
> 
> I see at least two possible improvements with the Chiffre d’Affaires API, that we should share with API Entreprise :
> 
> * we lack the bilan type (complet, simplifié etc.)
> * we dont have any CA older than 3 years ago
> 
> This revamp include pretty heavy changes to the open data page and I think we can communicate about it @HAEKADI ![image](https://private-user-images.githubusercontent.com/8900205/429999125-b8cb9db5-1af2-4e72-8909-e5f9290b7d91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwOTk5MTgsIm5iZiI6MTc0NDA5OTYxOCwicGF0aCI6Ii84OTAwMjA1LzQyOTk5OTEyNS1iOGNiOWRiNS0xYWYyLTRlNzItODkwOS1lNWY5MjkwYjdkOTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MDhUMDgwNjU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGNiMTgyNGQzZTI3MmVkOTBkYTM3NGMxZWIzZTc5ZmZhNzEwMWJiN2EwNTgxNDRhM2FiMTRlYTJlMWQ0NjI3MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.bPFlaoCn6Ml373V6d70kB394F0gBgpOMvAZHIXGA5SI)

Good job ! 👌

I fixed e2e tests, some typo and I hided the null points in the chart.
Ready to merge !
